### PR TITLE
List: Improve aria for collapsible groups

### PR DESCRIPTION
**Pull request by marker-dao**

### CODE REVIEW

File: packages/devextreme/js/__internal/ui/list/m_list.base.ts [hunk 1]
```diff
@@ -781,27 +786,67 @@ export const ListBase = CollectionWidget.inherit({
...
       .addClass(LIST_GROUP_HEADER_CLASS)
-      .attr('id', id)
+      .attr('id', groupHeaderId)
       .appendTo($groupElement);
 
-    const groupTemplateName = this.option('groupTemplate');
-    const groupTemplate = this._getTemplate(group.template || groupTemplateName, group, index, $groupHeaderElement);
+    const { groupTemplate: templateName } = this.option();
+
+    const groupTemplate = this._getTemplate(
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/__internal/ui/list/m_list.base.ts:843]: JFYI: we have alias to get template by option name directly
https://github.com/DevExpress/DevExtreme/blob/46b526fbcbdcdb33a32fada7688ae46f6f0273ef/packages/devextreme/js/__internal/core/widget/dom_component.ts#L562-L563

Comment (EugeniyKiyashko) [reply packages/devextreme/js/__internal/ui/list/m_list.base.ts:843]: https://github.com/DevExpress/DevExtreme/blob/46b526fbcbdcdb33a32fada7688ae46f6f0273ef/packages/devextreme/js/__internal/core/widget/dom_component.ts#L562-L563

**Review by EugeniyKiyashko — APPROVED** [review]

File: packages/devextreme/js/__internal/ui/list/m_list.base.ts [hunk 2]
```diff
@@ -781,27 +786,67 @@ export const ListBase = CollectionWidget.inherit({
...
+    this.setAria({ expanded }, $groupHeader);
+  },
+
+  _setGroupHeaderAria($groupHeader, listGroupBodyId): void {
+    const { collapsibleGroups } = this.option();
+
+    const groupHeaderAria = {
+      role: collapsibleGroups ? 'button' : undefined,
+      expanded: collapsibleGroups ? true : undefined,
+      controls: collapsibleGroups ? listGroupBodyId : undefined,
```
Comment (ksercs) [inline packages/devextreme/js/__internal/ui/list/m_list.base.ts:811]: [minor]
now "collapsibleGroups" is copied and pasted several times, i'd prefer clear `if` using

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/listParts/commonTests.js [hunk 3]
```diff
@@ -4485,4 +4485,183 @@ QUnit.module('Accessibility', () => {
...
+
+            const $groupElement = $element.find(`.${LIST_GROUP_CLASS}`);
+
+            const expectedGroupAria = {
+                role: collapsibleGroups ? 'group' : undefined,
+                'aria-labelledby': collapsibleGroups ? true : false,
+            };
+
+            assert.strictEqual($groupElement.attr('role'), expectedGroupAria.role, 'role is correct');
+            assert.strictEqual(!!$groupElement.attr('aria-labelledby'), expectedGroupAria['aria-labelledby'], 'aria-labelledby is correct');
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/listParts/commonTests.js:4603]: It'd be better to test real ids, don't see stoppers for this. 
Not it's difficult to read and can be broken (low probability)

File: packages/devextreme/js/__internal/ui/list/m_list.base.ts [hunk 4]
```diff
@@ -781,27 +786,67 @@ export const ListBase = CollectionWidget.inherit({
...
+    };
+
+    this.setAria(groupHeaderAria, $groupHeader);
+  },
+
+  _setGroupBodyAria($groupBody, groupHeaderId): void {
```
Comment (ksercs) [inline packages/devextreme/js/__internal/ui/list/m_list.base.ts:817]: [minor, ts]
It'd be cool if we fully type new functions

**Review by ksercs — APPROVED** [review]

File: packages/devextreme/js/__internal/ui/list/m_list.base.ts (hunk 2 above)
Comment (marker-dao) [reply packages/devextreme/js/__internal/ui/list/m_list.base.ts:811]: Right now yes, but I believe each of these attributes should be configurable separately. If you use ‘if’, you will configure all attributes at once.

**Pull request merged**